### PR TITLE
Venkat unittest

### PR DESCRIPTION
**Pull request by XLRA**
## Changes Made
This merge introduces unit tests for two key components, enhancing our application's reliability and maintainability. The tests cover the following:

1. **CallToAction Component Testing**: Verifies the presence of the main heading, description text, and the correct link path for the "Sign Up" button. Additionally, it checks for the presence and correctness of SVG elements within the component, ensuring that our UI renders as expected.

2. **ThemeToggler Component Testing**: Focuses on the functionality of the theme toggling feature. It tests the ability to switch themes from light to dark and vice versa, leveraging the `next-themes` package. Mocking of the `useTheme` hook allows for precise verification of theme changes triggered by user interactions.

## Checklist before requesting a review

- [✓] I have performed a self-review of my code.
- [✓] There are no lint nor build errors in my code.
- [✓] If it is a core feature, I have added thorough tests.

### CODE REVIEW

**Review by SukrutN — APPROVED** [review]
### Code Review Summary

#### General Observations:
- The test suites for both `CallToAction` and `ThemeToggler` components are well-structured and follow common testing patterns with React Testing Library and Jest.
- Use of `@testing-library/jest-dom` for more readable and powerful assertions is a good practice.
- Mocking `next-themes` for `ThemeToggler` tests is appropriately handled, allowing for isolation of component behavior.

#### Specific Suggestions:

##### CallToAction Component Tests
1. **Repeated Render Calls**: Each test case renders the `CallToAction` component independently. Consider using `beforeEach` for rendering to DRY (Don't Repeat Yourself) your code. This approach minimizes redundancy and makes the tests easier to maintain.
   
2. **Direct DOM Queries**: You're directly querying SVG elements via `document.querySelector` and `document.querySelectorAll`. While this works, it's encouraged to use RTL (React Testing Library) queries when possible to maintain consistency and utilize RTL's built-in functionalities.
   
3. **SVG Element Testing**: Verifying the presence of SVG elements by their `viewBox` attributes and counting specific children elements (like `circle` and `path`) is an unusual practice in high-level component tests. Ensure this closely aligns with your testing goals, focusing on user interactions and accessibility rather than internal structure unless specifically needed for visual regression testing purposes.

##### ThemeToggler Component Tests
1. **State Management in Tests**: The mocking strategy for `useTheme` is effective. However, consider adding a test to ensure that the initial theme state can be both `light` and `dark`, as this would more thoroughly test the toggler's behavior in different initial conditions.
   
2. **Button Accessibility**: The test uses `{ name: 'theme toggler' }` to query the button, which is good for accessibility. Ensure that your button element also includes accessible attributes in the component itself, such as `aria-label`, to support this.
   
3. **Test Coverage**: While basic functionality is tested, consider adding more scenarios if your component has more complex interactions or states. For example, verifying the visual feedback to the user upon theme change could make the tests more comprehensive.

#### Additional Notes:
- **Code Formatting**: The code is generally well-formatted. Consistent use of semicolons and single quotes vs. double quotes could further improve readability.
- **Test Descriptions**: Descriptions are clear and concise. They accurately reflect what each test aims to achieve, which is crucial for maintainability and understanding the purpose of each test case.

**Review by SukrutN — APPROVED** [review]
### Code Review Summary

#### General Observations:
- The test suites for both `CallToAction` and `ThemeToggler` components are well-structured and follow common testing patterns with React Testing Library and Jest.
- Use of `@testing-library/jest-dom` for more readable and powerful assertions is a good practice.
- Mocking `next-themes` for `ThemeToggler` tests is appropriately handled, allowing for isolation of component behavior.

#### Specific Suggestions:

##### CallToAction Component Tests
1. **Repeated Render Calls**: Each test case renders the `CallToAction` component independently. Consider using `beforeEach` for rendering to DRY (Don't Repeat Yourself) your code. This approach minimizes redundancy and makes the tests easier to maintain.
   
2. **Direct DOM Queries**: You're directly querying SVG elements via `document.querySelector` and `document.querySelectorAll`. While this works, it's encouraged to use RTL (React Testing Library) queries when possible to maintain consistency and utilize RTL's built-in functionalities.
   
3. **SVG Element Testing**: Verifying the presence of SVG elements by their `viewBox` attributes and counting specific children elements (like `circle` and `path`) is an unusual practice in high-level component tests. Ensure this closely aligns with your testing goals, focusing on user interactions and accessibility rather than internal structure unless specifically needed for visual regression testing purposes.

##### ThemeToggler Component Tests
1. **State Management in Tests**: The mocking strategy for `useTheme` is effective. However, consider adding a test to ensure that the initial theme state can be both `light` and `dark`, as this would more thoroughly test the toggler's behavior in different initial conditions.
   
2. **Button Accessibility**: The test uses `{ name: 'theme toggler' }` to query the button, which is good for accessibility. Ensure that your button element also includes accessible attributes in the component itself, such as `aria-label`, to support this.
   
3. **Test Coverage**: While basic functionality is tested, consider adding more scenarios if your component has more complex interactions or states. For example, verifying the visual feedback to the user upon theme change could make the tests more comprehensive.

#### Additional Notes:
- **Code Formatting**: The code is generally well-formatted. Consistent use of semicolons and single quotes vs. double quotes could further improve readability.
- **Test Descriptions**: Descriptions are clear and concise. They accurately reflect what each test aims to achieve, which is crucial for maintainability and understanding the purpose of each test case.

File: src/components/Header/ThemeToggler.test.tsx [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    return {
+        useTheme: jest.fn(() => ({
+            theme,
+            setTheme,
+        })),
+    }
+})
+
+describe('ThemeToggler', () => {
+  it('toggles theme from light to dark', () => {
```
Comment (ThatZiv) [inline src/components/Header/ThemeToggler.test.tsx:18]: Would be cool to see this on the main site hopefully one day :)

**Review by ThatZiv — APPROVED** [review]
All good!

**Pull request merged**